apis/datasciencecluster/v1/datasciencecluster_types.go

```diff
@@ -114,7 +108,7 @@ type DataScienceClusterStatus struct {
 
 	// Expose component's specific status
 	// +optional
-	Components ComponentsStatus `json:"components,omitempty"`
+	Components map[string]status.ReleaseStatus `json:"components,omitempty"`
```

Comment:
the `ComponentStatus` is meant to hold more information that the release status so I would not conflate different kind of information into a `ReleaseStaus` struct as it would be misleading. 

I would also recommend not to use a map as:
- the `ComponentStatus` should be symmetric to the `spec. Components` in term of naming 
- With a map, it would be extremely easy to accidental break the API